plotly/tools.py

```diff
@@ -1443,6 +1443,376 @@ class FigureFactory(object):
     """
 
     @staticmethod
+    def create_scatterplotmatrix(df, useindex = False, index = " ",
+                                 diagonal = "Scatter", symbol = 0, size = 6,
+                                 height = " ", width = " ", jitter = 0,
+                                 title = "Scatterplot Matrix"):
+
+        from plotly import tools
+        import plotly.plotly as py
+        import plotly.graph_objs as go
+        import pandas as pd
+        import random as r
+       
+        colors = ["rgb(31, 119, 180)", "rgb(255, 127, 14)",
+                  "rgb(44, 160, 44)", "rgb(214, 39, 40)",
+                  "rgb(148, 103, 189)", "rgb(140, 86, 75)",
+                  "rgb(227, 119, 194)", "rgb(127, 127, 127)",
+                  "rgb(188, 189, 34)", "rgb(23, 190, 207)"]
+
+        matrix = []
+        col_names = []
+        index_vals = []
+        diagonal_choices = ["Scatter", "Histogram"]
+
+        # Check if pandas dataframe
+        if type(df) != pd.core.frame.DataFrame:
```

Comment:
You should try to prefer `isInstance(df, pd.core.frame.DataFrame)` here.

https://stereochro.me/ideas/type-vs-isinstance